code/modules/assembly/signaler.dm

```diff
@@ -190,8 +191,35 @@
 	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
 	resistance_flags = FIRE_PROOF
+	eaten_text = "you've ate the anomaly core..."
 	var/anomaly_type = /obj/effect/anomaly
 
+/obj/item/assembly/signaler/anomaly/attack(mob/living/M, mob/living/user)
+	if(user.a_intent == INTENT_HARM || M != user || !ishuman(user))
+		return ..()
+
+	var/mob/living/carbon/human/H = user
+	var/obj/item/organ/stomach/S = H.getorganslot(ORGAN_SLOT_STOMACH)
+
+	if(!istype(S, /obj/item/organ/stomach/cell))//need a fancy stomach for it
+		return ..()
```

Comment:
probably shouldn't be based on stomach, should probably just be based on them being an ipc
if it's just based on stomach then ethereals and preterni would be able to eat them too